autopilot/contractor.go

```diff
@@ -91,36 +105,69 @@ func (c *contractor) performContractMaintenance() error {
 	return nil
 }
 
-func (c *contractor) runContractChecks() error {
+func (c *contractor) runContractChecks(cfg Config, blockHeight uint64) error {
 	// fetch all active contracts
-	active, err := c.ap.bus.ActiveContracts(math.MaxUint64)
+	active, err := c.ap.bus.ActiveContracts("", math.MaxUint64)
 	if err != nil {
 		return err
 	}
 
-	// TODO: check for IP range violations
+	// create an IP filter
+	ipFilter := newIPFilter()
 
-	// run checks on the contracts individually
+	// check every contract
 	for _, contract := range active {
-		// grab contract metadata
-		metadata := contract.Metadata
-
-		// update metadata:
-		// TODO: is host in host DB
-		// TODO: is max revision
-		// TODO: is offline
-		// TODO: is gouging
-		// TODO: is low score
-		// TODO: is out of funds
-
-		// TODO: is up for renewal
-		// TODO: is GFU limited (allowance # hsots)
-
-		// update contract metadata
-		err = c.ap.bus.UpdateContractMetadata(contract.ID, metadata)
+		// fetch host from hostdb
+		host, err := c.host(contract)
+		if err != nil {
+			return err
+		}
+
+		// fetch metadata
+		metadata, err := c.ap.bus.ContractMetadata(contract.ID)
 		if err != nil {
 			return err
 		}
+
+		// apply host filter
+		newMetadata, _, updated := newHostFilter(host, metadata).
+			withBlackListFilter(cfg).
+			withMaxRevisionFilter().
+			withOfflineFilter().
+			withRedundantIPFilter(ipFilter).
+			withRemainingFundsFilter(cfg).
+			// TODO: score filter
+			withUpForRenewalFilter(cfg, blockHeight).
+			withWhiteListFilter(cfg).
+			finalize()
+
+		// apply update
+		if updated {
+			err = c.ap.bus.UpdateContractMetadata(contract.ID, newMetadata)
```

Comment:
Seems like `c` only has `ap` so that it can access `bus`. Consider giving `c` a `bus` field directly instead.

Reply:
It uses the `bus`, `store` and methods defined on the `ap` itself. That said, I agree the contractor should probably only have a `Bus` and everything else should be defined on the contractor or passed in. I'll move it in a separate PR since it's quite a bit of code moving that's not related to host filter.